# **Data Preprocessing**

**Project:** Airbnb Pricing Analysis

**Team:** Jessie Dicker, Norma Arredondo

**Date:** March 22, 2026




## Table Contents
1. Data Load
2. Drop Rows with Missing TARGET
3. Simple Cleaning
4. Train-Test Split
5. Handle Missing Values
6. Handle Outliers
7. Key Finding Summary

## 1. Load Data

In [15]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Settings
plt.style.use('seaborn-v0_8')
pd.set_option('display.max_columns', None)

print("✓ Libraries loaded!")

✓ Libraries loaded!


In [16]:
# Load your data
df = pd.read_csv('../data/raw/AB_NYC_2019.csv')
df.head()

,id,name,host_id,host_name,neighbourhood_group,neighbourhood,latitude,longitude,room_type,price,minimum_nights,number_of_reviews,last_review,reviews_per_month,calculated_host_listings_count,availability_365
0,2539,Clean & quiet apt home by the park,2787,John,Brooklyn,Kensington,40.64749,-73.97237,Private room,149,1,9,2018-10-19,0.21,6,365
1,2595,Skylit Midtown Castle,2845,Jennifer,Manhattan,Midtown,40.75362,-73.98377,Entire home/apt,225,1,45,2019-05-21,0.38,2,355
2,3647,THE VILLAGE OF HARLEM....NEW YORK !,4632,Elisabeth,Manhattan,Harlem,40.80902,-73.94190,Private room,150,3,0,NaN,NaN,1,365
3,3831,Cozy Entire Floor of Brownstone,4869,LisaRoxanne,Brooklyn,Clinton Hill,40.68514,-73.95976,Entire home/apt,89,1,270,2019-07-05,4.64,1,194
4,5022,Entire Apt: Spacious Studio/Loft by central park,7192,Laura,Manhattan,East Harlem,40.79851,-73.94399,Entire home/apt,80,10,9,2018-11-19,0.10,1,0


In [17]:
# Data Shape
df.shape

(48895, 16)

In [18]:
# Data Frame information
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

### Load Data: Questions

- How many rows and columns does your raw dataset have?

There are 48,895 rows and 16 columns in the raw dataset.

- Which columns are your target variable?

The only target variable in this project is 'price'.

- What issues from Notebook 01 will you address in this notebook?

In this notebook, we will address the value of the impossible values in price and availability. As well as handling missing values and outliers.

## 2. Drop Rows with Missing TARGET

In [19]:
target_col = 'price'
missing_target = df[target_col].isnull().sum()


print(f"\n Mising values in target '{target_col}' : {missing_target}")
print(f"Percentage:  {missing_target/len(df)*100:.2f}%")


 Mising values in target 'price' : 0
Percentage:  0.00%


In [20]:
# DROP rows with missing target
df_clean = df.dropna(subset=['price'])

In [21]:
# Verify no missing targets remain

df_clean[target_col].isnull().sum()

np.int64(0)

In [22]:
# Check missing value in df_clean in features

df_clean.isnull().sum()

id                                    0
name                                 16
host_id                               0
host_name                            21
neighbourhood_group                   0
neighbourhood                         0
latitude                              0
longitude                             0
room_type                             0
price                                 0
minimum_nights                        0
number_of_reviews                     0
last_review                       10052
reviews_per_month                 10052
calculated_host_listings_count        0
availability_365                      0
dtype: int64

In [23]:
df_clean.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

### Drop Rows with Missing TARGET: Questions
- How many rows had a missing target value?

There are 0 rows with missing values in our target variable, price.
  
- What percentage of the dataset does that represent?

It represent 0.00% of the dataset.

- How many rows remain after dropping?

There are still 48,895 rows remaining after dropping.

## 3. Simple Cleaning

In [24]:
# Check for duplicates
duplicates = df_clean.duplicated().sum()
print(f'\n Duplicate rows found: {duplicates}')


 Duplicate rows found: 0


In [25]:
# Remove duplicates
if duplicates>0:
    df_clean = df_clean.drop_duplicates()
    print(f'removed {duplicates} duplicated rows')
else:
    print('No duplicates removed')

No duplicates removed


In [26]:
# check impossible values
positive_col = ['price', 'minimum_nights', 'number_of_reviews', 
                'reviews_per_month', 'calculated_host_listings_count', 
                'availability_365']
for col in positive_col:
    neg_count = (df_clean[col]<0).sum()
    if neg_count:
        print(f'{col}: has negative values')

        ## remove negative values
        df_clean = df_clean[df_clean[col]>=0]

        print(f'removed {neg_count} rows with negative {col}')

In [27]:
# Remove price = 0 
zero_price = (df_clean['price'] == 0).sum()
print(f'\nRows with Price = $0: {zero_price}')

df_clean = df_clean[df_clean['price'] > 0]
print(f'Removed {zero_price} rows with price = $0')

# Remove availability_365 more than 365
avail_365 = (df_clean['availability_365'] > 365).sum()
print(f'\nRows with more than 365: {avail_365}')

if avail_365:
    df_clean = df_clean[df_clean['availability_365'] <= 365]
    print(f'Removed {avail_365} rows with availability > 365')


Rows with Price = $0: 11
Removed 11 rows with price = $0

Rows with more than 365: 0


In [28]:
# Drop data leakage
leakage_cols = ['id', 'name', 'host_id', 'host_name', 'last_review']

df_clean = df_clean.drop(columns=leakage_cols)
print(f'Dropped columns: {leakage_cols}')
print(f'Remaining columns: {df_clean.columns.tolist()}')

Dropped columns: ['id', 'name', 'host_id', 'host_name', 'last_review']
Remaining columns: ['neighbourhood_group', 'neighbourhood', 'latitude', 'longitude', 'room_type', 'price', 'minimum_nights', 'number_of_reviews', 'reviews_per_month', 'calculated_host_listings_count', 'availability_365']


In [29]:
# Verify no more impossible values

positive_col = ['price', 'minimum_nights', 'number_of_reviews', 
                'reviews_per_month', 'calculated_host_listings_count', 
                'availability_365']

issue_found = False
for col in positive_col:
    neg_count = (df_clean[col]<0).sum()
    if neg_count:
        print(f'{col}: has negative values')
        issue_found = True

if not issue_found:
    print('No impossible values remaining')

No impossible values remaining


In [32]:
print(f"\nDataset shape originally: {df.shape}")
print(f"\nDataset shape after all pre-split cleaning: {df_clean.shape}")


Dataset shape originally: (48895, 16)

Dataset shape after all pre-split cleaning: (48884, 11)


### Cleaning Notes
- After duplicates removal, 48,895 rows remain because there are no duplicates. 
- Impossible Values were removed, including price equal to and less than 0, because it is not a realistic listing, and rows with more than 365 in availability_365, because it should exceed 365 days.
- Data leakage columns dropped:

  id - A unique identifier, and it's not a predictive value

  name - Not a structured feature

  host_id - It is an arbitrary numerical identifier

  host_name - name is not a predictive value

  last_review - this column would leak temporal information
- After removing duplicates, impossible values, and data leakage, there are 48,884 rows and 11 columns left.

## 4. Train-Test Slpit

## 5. Handle Missing Values

## 6. Handle Outliers

## Key Findings Summary